be/src/exprs/math_functions.cpp

```diff
@@ -687,18 +687,11 @@ StatusOr<ColumnPtr> MathFunctions::conv_string(FunctionContext* context, const C
     return result.build(ColumnHelper::is_all_const(columns));
 }
 
-static uint32_t generate_randoms(ColumnBuilder<TYPE_DOUBLE>* result, int32_t num_rows, uint32_t seed) {
-    for (int i = 0; i < num_rows; ++i) {
-        seed = ::rand_r(&seed);
-        // Normalize to [0,1].
-        result->append(static_cast<double>(seed) / RAND_MAX);
-    }
-    return seed;
-}
-
+std::uniform_real_distribution<double>* MathFunctions::_distribution =
+        new std::uniform_real_distribution<double>(lower_bound, upper_bound);
 Status MathFunctions::rand_prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
     if (scope == FunctionContext::THREAD_LOCAL) {
-        int64_t seed = 0;
+        std::mt19937_64* generator = new std::mt19937_64();
```

Comment:
replace with static thread_local std::mt19937 generator;

Reply:
here was my test result after this change:
![image](https://github.com/StarRocks/starrocks/assets/34912776/e4411b25-e1d1-4fcf-ab00-aaef86c35903)

Reply:
Thanks, I'll try to modify and test it